apis/python/node/src/lib.rs

```diff
@@ -85,16 +39,42 @@ impl Node {
     pub fn send_output(
         &mut self,
         output_id: String,
-        data: &PyBytes,
+        data: PyObject,
+        metadata: Option<&PyDict>,
+        py: Python,
+    ) -> eyre::Result<()> {
+        if let Ok(py_bytes) = data.downcast::<PyBytes>(py) {
+            let data = py_bytes.as_bytes();
+            self.send_output_slice(output_id, data.len(), data, metadata)
+        } else if let Ok(arrow_array) = arrow::array::ArrayData::from_pyarrow(data.as_ref(py)) {
```

Comment:
I think it would be great to make sure that the datatype is `Uint8` as currently passing other types does not raise any errors.